core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/constraintcomponents/AndConstraintComponent.java

```diff
@@ -58,13 +58,14 @@ public AndConstraintComponent(AndConstraintComponent andConstraintComponent) {
 
 	@Override
 	public void toModel(Resource subject, IRI predicate, Model model, Set<Resource> exported) {
-		model.add(subject, SHACL.AND, getId());
-		HelperTool.listToRdf(and.stream().map(Shape::getId).collect(Collectors.toList()), getId(), model);
-
 		if (exported.contains(getId())) {
 			return;
 		}
 		exported.add(getId());
+
+		model.add(subject, SHACL.AND, getId());
+		HelperTool.listToRdf(and.stream().map(Shape::getId).collect(Collectors.toList()), getId(), model);
+
```

Comment:
Found an issue when there were multiple violations for the same shape and that shape used an RDF list. Essentially the list would be printed multiple times, but would have the same head node and thus become corrupt. This code here is part of that solution.